features/strict_mode.feature

```diff
@@ -0,0 +1,18 @@
+Feature: Reporting Strict Mode Exceptions
+
+Scenario: StrictMode DiscWrite violation
+    When I run "StrictModeDiscScenario" with the defaults
+    Then I should receive a request
+    And the request is a valid for the error reporting API
+    And the exception "errorClass" equals "android.os.StrictMode$StrictModeViolation"
+    And the exception "message" equals "policy=262145 violation=1"
+    And the event "metaData.StrictMode.Violation" equals "DiskWrite"
+
+
+Scenario: StrictMode Network on Main Thread violation
+    When I run "StrictModeNetworkScenario" with the defaults
+    Then I should receive a request
+    And the request is a valid for the error reporting API
+    And the exception "errorClass" equals "android.os.StrictMode$StrictModeViolation"
+    And the exception "message" equals "policy=262148 violation=4"
+    And the event "metaData.StrictMode.Violation" equals "NetworkOperation"
```

Comment:
I occasionally get this:

```
  Scenario: StrictMode Network on Main Thread violation                               # features/strict_mode.feature:12
    When I run "StrictModeNetworkScenario" with the defaults                          # features/steps/build_steps.rb:1
    Then I should receive a request                                                   # maze-runner-65bdcb7a135e/lib/features/steps/request_assertion_steps.rb:11
    And the request is a valid for the error reporting API                            # maze-runner-65bdcb7a135e/lib/features/steps/error_reporting_steps.rb:1
      the field 'events' must be a non-empty array (Minitest::Assertion)
      features/strict_mode.feature:15:in `And the request is a valid for the error reporting API'
    And the exception "errorClass" equals "android.os.StrictMode$StrictModeViolation" # maze-runner-65bdcb7a135e/lib/features/steps/error_reporting_steps.rb:47
    And the exception "message" equals "policy=262148 violation=4"                    # maze-runner-65bdcb7a135e/lib/features/steps/error_reporting_steps.rb:47
    And the event "metaData.StrictMode.Violation" equals "NetworkOperation"           # maze-runner-65bdcb7a135e/lib/features/steps/error_reporting_steps.rb:25
```

Request output:

```
URI: http://10.0.2.2:9291/
HEADERS:
  content-type: 'application/json'
  bugsnag-sent-at: '2018-03-15T10:40:21Z'
  bugsnag-api-key: 'a35a2a72bd230ac0aa0f52715bbdc6aa'
  bugsnag-payload-version: '4.0'
  user-agent: 'Dalvik/2.1.0 (Linux; U; Android 7.0; Android SDK built for x86_64 Build/NYC)'
  host: '10.0.2.2:9291'
  connection: 'Keep-Alive'
  accept-encoding: 'gzip'

BODY:
{
  "apiKey": "a35a2a72bd230ac0aa0f52715bbdc6aa",
  "payloadVersion": "4.0",
  "notifier": {
    "name": "Android Bugsnag Notifier",
    "version": "4.3.1",
    "url": "https://bugsnag.com"
  },
  "events": [

  ]
}
```

Any idea what's going on here?

Reply:
Hm, not 100% sure, but could this be another manifestation of #270? If the errorFile is deleted prior to request serialisation then an empty array of events would be expected. Is only 1 request received by this scenario?

Reply:
(I tried running this around 5 times and it seemed to pass consistently on my setup)

Reply:
It seems to be inconsistent and could potentially be related to #270 - there seems to exist a case where the payload is sent but no events are present.